dart/lib/src/event_processor/exception/io_exception_event_processor.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:io';
+
+import '../../protocol.dart';
+import 'exception_event_processor.dart';
+
+ExceptionEventProcessor exceptionEventProcessor() =>
+    IoExceptionEventProcessor();
+
+class IoExceptionEventProcessor implements ExceptionEventProcessor {
+  @override
+  SentryEvent apply(SentryEvent event, {dynamic hint}) {
+    final throwable = event.throwable;
+    if (throwable is HttpException) {
+      return applyHttpException(throwable, event);
+    }
+    if (throwable is SocketException) {
+      return applySocketException(throwable, event);
+    }
+    if (throwable is FileSystemException) {
+      return applyFileSystemException(throwable, event);
+    }
+
+    return event;
+  }
+
+  // https://api.dart.dev/stable/dart-io/HttpException-class.html
+  SentryEvent applyHttpException(HttpException exception, SentryEvent event) {
```

Comment:
```suggestion
  SentryEvent _applyHttpException(HttpException exception, SentryEvent event) {
```

Also for the other methods, or is there a reason to be public?

Reply:
Nope, just an oversight from me.